### DISCUSSION TITLE
Don't use NPM; use Yarn

### DISCUSSION THREAD

**Discussion by ghost** [uNetworking/uWebSockets.js#413] [General]
NPM has been one of the biggest irritations since day 1 of this project. I hate pretty much everything NPM is about, does, and how their incompetent personnel don't care about anything related to native modules. Their search is the worst I have ever seen, their support sucks, everything about them is lousy and half-assed.

With NPM 7 they have broken our installation process; you can no longer upgrade to a new package version without first manually uninstalling it and then reinstalling it again. They have poor error messages and none of this is clear to the user.

Further on NPM is written by incompetent morons and is taking 24 seconds to install a new version, while Yarn does it in 4 seconds or even 2 seconds - and it actually works like it should - it does install the latest version when you tell it to.

So that's why I'm no longer going to care about NPM whatsoever. We don't need them, they can go do whatever they want.

Use Yarn, it works and is 5x faster. Don't post issues about NPM - I don't care.

**Comment by Jak2k** [comment]
pnpm

**Comment by uNetworkingAB** [comment]
Anything that is not NPM will do. You can't do worse than NPM.

**Reply by porsager** [reply]
`bun install` works great 👍
